Build related, license-maven-plugin fails when running the build from a child module

build related issue and relatively minor at that. may not be worth fixing but it is kind of annoying. Good thing oshi is a relatively small project

````
git clone https://github.com/oshi/oshi.git
mvn clean install
cd oshi-core
mvn clean install
````

error message
````
[ERROR] Failed to execute goal com.mycila:license-maven-plugin:3.0:format (default) on project oshi-core: Execution default of goal com.mycila:license-maven-plugin:3.0:format failed: Cannot read header document LICENSE_HEADER. Cause: Resource LICENSE_HEADER not found in file system, classpath or URL: no protocol: LICENSE_HEADER -> [Help 1]
````

--- a/oshi-core-shaded/pom.xml
+++ b/oshi-core-shaded/pom.xml
@@ -40,6 +40,10 @@
 		<tag>HEAD</tag>
 	</scm>
 
+	<properties>
+		<main.basedir>${project.parent.basedir}</main.basedir>
+	</properties>
+
 	<dependencies>
 		<dependency>
 			<groupId>org.slf4j</groupId>

--- a/oshi-core/pom.xml
+++ b/oshi-core/pom.xml
@@ -42,6 +42,7 @@
 
 	<properties>
 		<jna.version>5.1.0</jna.version>
+		<main.basedir>${project.parent.basedir}</main.basedir>
 	</properties>
 
 	<dependencies>

--- a/oshi-dist/pom.xml
+++ b/oshi-dist/pom.xml
@@ -38,6 +38,10 @@
 		<tag>HEAD</tag>
 	</scm>
 
+	<properties>
+		<main.basedir>${project.parent.basedir}</main.basedir>
+	</properties>
+
 	<!-- NOTE: These dependency declarations are only required to sort this 
 		project to the end of the line in the multimodule build. -->
 	<dependencies>

--- a/oshi-json/pom.xml
+++ b/oshi-json/pom.xml
@@ -40,6 +40,10 @@
 		<tag>HEAD</tag>
 	</scm>
 
+	<properties>
+		<main.basedir>${project.parent.basedir}</main.basedir>
+	</properties>
+
 	<dependencies>
 		<dependency>
 			<groupId>${project.groupId}</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -147,7 +147,7 @@
 		<maven-dependency-plugin.version>3.1.1</maven-dependency-plugin.version>
 		<maven-release-plugin.version>2.5.3</maven-release-plugin.version>
 		<maven-source-plugin.version>3.0.1</maven-source-plugin.version>
-		<spotbugs-maven-plugin.version>3.1.8</spotbugs-maven-plugin.version>
+		<spotbugs-maven-plugin.version>3.1.9</spotbugs-maven-plugin.version>
 		<jacoco-maven-plugin.version>0.8.2</jacoco-maven-plugin.version>
 		<coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>
 		<tidy-maven-plugin.version>1.1.0</tidy-maven-plugin.version>
@@ -164,7 +164,7 @@
 		<maven-scm-plugin.version>1.11.1</maven-scm-plugin.version>
 		<maven-scm-provider-jgit.version>1.11.1</maven-scm-provider-jgit.version>
 		<JavaEWAH.version>1.1.6</JavaEWAH.version>
-		<jsch.version>0.1.54</jsch.version>
+		<jsch.version>0.1.55</jsch.version>
 		<jgit.version>5.1.3.201810200350-r</jgit.version>
 		<plexus-utils.version>3.1.0</plexus-utils.version>
 		<!-- Misc. -->
@@ -172,7 +172,7 @@
 		<wagon-maven-plugin.version>2.0.0</wagon-maven-plugin.version>
 		<animal-sniffer-maven-plugin.version>1.17</animal-sniffer-maven-plugin.version>
 		<dependency-check-maven.version>3.3.4</dependency-check-maven.version>
-		<puppycrawl.checkstyle.version>8.14</puppycrawl.checkstyle.version>
+		<puppycrawl.checkstyle.version>8.15</puppycrawl.checkstyle.version>
 		<m2e.lifecycle-mapping.version>1.0.0</m2e.lifecycle-mapping.version>
 	</properties>
 

--- a/pom.xml
+++ b/pom.xml
@@ -110,6 +110,8 @@
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 		<signature.artifact>java17</signature.artifact>
 		<signature.version>1.0</signature.version>
+		<!-- Base directory, overwrite in submodules -->
+		<main.basedir>${project.basedir}</main.basedir>
 		<!-- Dependency versions -->
 		<slf4j.version>1.7.25</slf4j.version>
 		<junit-vintage-engine.version>4.12.3</junit-vintage-engine.version>
@@ -484,7 +486,7 @@
 					<artifactId>license-maven-plugin</artifactId>
 					<version>${license-maven-plugin.version}</version>
 					<configuration>
-						<header>LICENSE_HEADER</header>
+						<header>${main.basedir}/LICENSE_HEADER</header>
 						<excludes>
 							<exclude>**/*test.*.txt</exclude>
 							<exclude>**/maven-wrapper.properties</exclude>

